app/src/main/res/layout/fragment_pin.xml

```diff
@@ -6,17 +6,44 @@
     android:layout_height="match_parent"
     tools:context=".features.pin.ui.PinFragment">
 
+    <TextView
+        android:id="@+id/pin_title"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:textSize="20sp"
+        android:textColor="?attr/textColorPrimary"
+        app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        app:layout_constraintBottom_toTopOf="@id/pinProgressView"
+        tools:text="@string/pin_enter_title"/>
+
+
     <com.yterletskyi.happyfriend.features.pin.ui.PinProgressView
         android:id="@+id/pinProgressView"
         android:layout_width="wrap_content"
         android:layout_height="wrap_content"
         android:layout_margin="16dp"
         app:defaultCircleColor="@color/mercury"
+        app:layout_constraintVertical_bias="0.1"
         app:layout_constraintEnd_toEndOf="parent"
         app:layout_constraintStart_toStartOf="parent"
         app:layout_constraintTop_toTopOf="parent"
+        app:layout_constraintBottom_toBottomOf="parent"
         app:progressCircleColor="?attr/colorPrimary" />
 
+    <TextView
+        android:id="@+id/pin_error"
+        android:layout_width="wrap_content"
+        android:layout_height="wrap_content"
+        android:textSize="12sp"
+        android:layout_marginTop="10dp"
+        android:textColor="@color/red"
```

Comment:
Let's please use `?attr/colorError` for errors. Also, please define it in `themes.xml` and `night/themes.xml` like we have other theme colors for exmaple `colorPrimary` etc.

There is a good article about meterial theme and colors: https://medium.com/over-engineering/setting-up-a-material-components-theme-for-android-fbf7774da739